storage-reporter/src/main/java/org/duracloud/mill/storagereporter/StorageReporter.java

```diff
@@ -0,0 +1,199 @@
+                                                  JpaSpaceStatsRepo.INTERVAL_DAY);
+                long total = 0;
+                if (stats != null && stats.size() > 0) {
+                    total = ((BigDecimal) stats.get(stats.size() - 1)[3])
+                            .longValue();
+                }
+                result.addStorageProviderResult(storageProviderAccount, total);
+            }
+
+            if (result.isOversubscribed()) {
+                oversubscribedAccounts.add(result);
+            } else {
+                undersubscribedAccounts.add(result);
+            }
+
+        }
+
+        // build subject
+        String subject = MessageFormat.format(
+                                              "Storage Report:  {0} Oversubscribed account(s)",
+                                              oversubscribedAccounts.size());
+
+        StringBuilder body = new StringBuilder();
+
+        if (oversubscribedAccounts.size() == 0) {
+            body.append("Presently there are no oversubscribed accounts.\n\n");
+        } else {
+            body.append("Oversubscribed Accounts: \n\n");
+        }
+        // write oversubscribed in initial block
+        for (AccountStorageReportResult result : oversubscribedAccounts) {
+            appendResultToBody(result, body);
+        }
+
+        body.append("\nAll other (ie undersubscribedAccounts) accounts: \n\n");
+
+        // write undersubscribedAccounts in following block
+        for (AccountStorageReportResult result : undersubscribedAccounts) {
+            appendResultToBody(result, body);
+        }
+
+        LOGGER.info("sending notification...");
+
+        // send email
+        notification.sendEmail(subject, body.toString());
+
+        LOGGER.info("Report complete: {} oversubscribed,  {} undersubscribedAccounts.",
+                    oversubscribedAccounts.size(),
+                    undersubscribedAccounts.size());
+
+        return new StorageReportResult(oversubscribedAccounts,
+                                       undersubscribedAccounts);
+    }
+
+    /**
+     * @param result
+     * @param body
+     */
+    private void appendResultToBody(AccountStorageReportResult result,
+                                    StringBuilder body) {
+        AccountInfo account = result.getAccount();
+        body.append(MessageFormat.format("{0} (subdomain={1}):\n",
+                                         account.getAcctName(),
+                                         account.getSubdomain()));
+        for (StorageProviderResult presult : result
+                .getStorageProviderResults()) {
+            StorageProviderAccount spa = presult.getStorageProviderAccount();
+            int limit = spa.getStorageLimit();
+            String total = new BigDecimal(presult.getTotalBytes())
+                    .divide(new BigDecimal(StorageProviderResult.TB),
+                            2,
+                            BigDecimal.ROUND_HALF_UP)
```

Comment:
The javadocs for BigDecimal suggest using RoundingMode.HALF_UP rather than BigDecimal.ROUND_HALF_UP